### PR TITLE
Fixes contribution with unblinded tokens

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7608

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
test plan defined in https://github.com/brave/brave-browser/issues/7608

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc [hunk 1]
```diff
@@ -157,7 +157,7 @@ void Unblinded::OnUnblindedTokens(
       continue;
     }
 
-    if (item->value + current_amount > reconcile.fee) {
+    if (current_amount >= reconcile.fee) {
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: this is the fix as we decided to go over if needed

File: components/brave_rewards/browser/database/database_unblinded_token.cc [hunk 2]
```diff
@@ -139,7 +148,8 @@ ledger::UnblindedTokenList DatabaseUnblindedToken::GetAllRecords(
     info->public_key = statement.ColumnString(2);
     info->value = statement.ColumnDouble(3);
     info->promotion_id = statement.ColumnString(4);
-    info->expires_at = statement.ColumnInt64(5);
+    info->expires_at =
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/database/database_unblinded_token.cc]: this fixes ads grants as some may expired, but we don't want to delete them

File: components/brave_rewards/browser/database/database_unblinded_token.cc [hunk 3]
```diff
@@ -77,10 +94,57 @@ bool DatabaseUnblindedToken::CreateTable(sql::Database* db) {
...
+  return true;
+}
+
+bool DatabaseUnblindedToken::MigrateToV14(sql::Database* db) {
+  DCHECK(db);
+  if (!db) {
+    return false;
+  }
+
+  const std::string query = base::StringPrintf(
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/database/database_unblinded_token.cc]: this fixes users in bad state

**Comment by NejcZdovc** [comment]
macos failed on `audit-network` and `dist`, re-running it now

**Comment by kjozwiak** [comment]
Reproduced the original issue using the following build:

Brave | 1.5.10 Chromium: 79.0.3945.88 (Official Build) nightly (64-bit)
--- | ---
Revision | c2a58a36b9411c80829b4b154bfcab97e581f1f3-refs/branch-heads/3945@{#954}
OS | macOS Version 10.15.2 (Build 19C57)

Verification **PASSED** using the following build:

Brave | 1.5.22 Chromium: 79.0.3945.117 (Official Build) nightly (64-bit)
Revision | 04f0a055010adab4484f7497fbfdbf312c307f1d-refs/branch-heads/3945@{#1019}
OS | macOS Version 10.15.2 (Build 19C57)

Case #1:

* launch Brave using `--enable-logging=stderr --rewards=staging=false`
* restore a wallet that currently has an ad grant (document in the shared GDrive)
* accept the UGP grant
* send 5 BAT to https://duckduckgo.com/ and ensure it goes through without any issues
* send 5 BAT to https://laurenwags.github.io/ and ensure it goes through without any issues
* check `brave://rewards` and ensure that everything is being displayed as expected

Case #2:

* launch Brave using `--enable-logging=stderr --rewards=staging=false,reconcile-interval=5`
* set `Monthly Payment` to `10 BAT`
* restore a wallet that currently has an ad grant (document in the shared GDrive)
* accept the UGP grant
* spend ~2min on https://duckduckgo.com/ and ensure it's being displayed in the a-c table
* spend ~2min on https://laurenwags.github.io/ and ensure it's being displayed in the a-c table
* ensure that a-c goes through and the correct amount is deducted from the total

Screen Shot 2020-01-09 at 4 05 49 PM
Screen Shot 2020-01-09 at 3 52 49 PM

**Comment by kjozwiak** [comment]
@LaurenWags noticed that there was a discrepancy in the amount of BAT that was being deducted from the account when tipping. @NejcZdovc mentioned that the server side fix hasn't been pushed. Waiting on @evq to fix the above server side. Once the fix has been pushed, @LaurenWags and myself will go through the cases outlined via https://github.com/brave/brave-core/pull/4317#issuecomment-572756624.

If things are looking good, we'll continue with the uplift process for both https://github.com/brave/brave-core/pull/4341 & https://github.com/brave/brave-core/pull/4342.

**Comment by kjozwiak** [comment]
As per earlier meetings, the a-c cases will be checked via https://github.com/brave/brave-browser/issues/7709.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.2 x64` using the following build:

Brave | 1.5.27 Chromium: 79.0.3945.117 (Official Build) nightly (64-bit)
-- | --
Revision | 04f0a055010adab4484f7497fbfdbf312c307f1d-refs/branch-heads/3945@{#1019}
OS | macOS Version 10.15.2 (Build 19C57)

* went through the cases outlined under https://github.com/brave/brave-core/pull/4317#issuecomment-572756624

Screen Shot 2020-01-12 at 11 07 04 PM

Screen Shot 2020-01-13 at 1 31 33 PM

**Pull request merged**